2_adventures/001/lenn4rd/03_raindrops_exercise/raindrops.rb

```diff
@@ -0,0 +1,25 @@
+class Fixnum
+  def prime_factor_of?(number)
+    number % self == 0
+  end
+end
+
+class Raindrops
+  PRIME_FACTORS = [3, 5, 7]
+
+  SOUNDS_FOR_FACTORS = {
+    '3' => 'Pling',
+    '5' => 'Plang',
+    '7' => 'Plong'
+  }
+  
+  def self.convert(number)
+    PRIME_FACTORS.each_with_object(sound = '') do |factor|
+      sound << SOUNDS_FOR_FACTORS[factor.to_s] if factor.prime_factor_of?(number)
+    end
+    
+    return number.to_s if sound.empty?
+    
+    sound
```

Comment:
Thanks for your feedback! I always have mixed feelings using the ternary operator. However, in this case it's probably the one that's easiest to read. I couldn't believe that your second statement would work because I was thinking about their `true` and `false` values only but since in a method Ruby always returns the last statement, it's `number` if `sound` is an empty string. Because this (at least for me) isn't self-explanatory and harder to understand, I'd go with the ternary operator here.